syft/pkg/cataloger/javascript/parse_node_binary.go

```diff
@@ -0,0 +1,38 @@
+package javascript
```

Comment:
I wonder if this belongs in something more like `syft/pkg/cataloger/binary/node.go` or similar, since it's not _exactly_ javascript but rather a mix of lots of C and some javascript.

Reply:
I organized it this way since when we're trying to identify an ecosystem-specific tool I believe it's easier to reason about the coverage of that grouping without spreading it too far across other packages:

- Example if we wanted to discover vulnerable versions of `npm` it would also be `package javascript`
- Lookups for things like `cargo, rustc` would fall under `package rust` 

Also see weston's comment here for the rust example
https://github.com/anchore/syft/pull/1296#discussion_r1008656419

Reply:
Definitely no strong opinion here... organizing by _ecosystem_ seems to make sense. 👍